tests/app/main/test_login.py

```diff
@@ -305,14 +409,724 @@ def test_reset_password_form_and_inputs_not_autofillable(
+                },
+                self.app.config['SHARED_EMAIL_KEY'],
+                self.app.config['INVITE_EMAIL_SALT']
+            )
+
+            res = self.client.get(
+                '/suppliers/create-user/{}'.format(token)
+            )
+
+            assert_equal(res.status_code, 200)
+            assert_true(
+                "Supplier Name"
+                in res.get_data(as_text=True)
+            )
+            assert_true(
+                '<button class="button-save">Create contributor account</button>'
+                in res.get_data(as_text=True)
+            )
+            assert_true(
+                '<form autocomplete="off" action="/suppliers/update-user/{}" method="POST" id="updateUserForm">'.format(token)  # noqa
+                in res.get_data(as_text=True)
+            )
+
+    def test_should_be_an_error_if_invalid_token_on_submit(self):
+        with self.app.app_context():
+            res = self.client.post(
+                '/suppliers/create-user/invalidtoken',
+                data={
+                    'password': '123456789',
+                    'name': 'name',
+                    'email_address': 'valid@test.com'}
+            )
+
+            assert_equal(res.status_code, 400)
+            assert_equal(
+                "Check you&#146;ve entered the correct link or ask the person who invited you to send a new invitation." in res.get_data(as_text=True),  # noqa
+                True
+            )
+            assert_equal(
+                '<button class="button-save">Create contributor account</button>' in res.get_data(as_text=True),
+                False
+            )
+
+    def test_should_be_an_error_if_invalid_token_on_update(self):
+        with self.app.app_context():
+            res = self.client.post(
+                '/suppliers/update-user/invalidtoken'
+            )
+
+            assert_equal(res.status_code, 400)
+            assert_equal(
+                "Check you&#146;ve entered the correct link or ask the person who invited you to send a new invitation." in res.get_data(as_text=True),  # noqa
+                True
+            )
+            assert_equal(
+                '<button class="button-save">Update user</button>' in res.get_data(as_text=True),  # noqa
+                False
+            )
+
+    def test_should_be_an_error_if_missing_name_and_password(self):
+        with self.app.app_context():
+
+            token = generate_token(
+                {
+                    'supplier_id': 1234,
+                    'supplier_name': 'Supplier Name',
+                    'email_address': 'testme@email.com'
+                },
+                self.app.config['SHARED_EMAIL_KEY'],
+                self.app.config['INVITE_EMAIL_SALT']
+            )
+
+            res = self.client.post(
+                '/suppliers/create-user/{}'.format(token),
+                data={
+                    'password': None,
+                    'name': None
+                }
```

Comment:
I don't think this would ever happen. Either the password and name are not submitted, in which case the data dictionary would be empty, or they are submitted empty, in which case they would have empty strings as values.